contrib/slait/slait.go

```
@@ -0,0 +1,227 @@
+				}
+			}
+		}
+	}
+	return err
+}
+
+func (ss *SlaitSubscriber) publicationToCSM(p cache.Publication) (*io.ColumnSeriesMap, error) {
+	columns := make(map[string]interface{}, len(ss.shape))
+	length := p.Entries.Len()
+	for _, shape := range ss.shape {
+		switch shape.Type {
+		case io.INT32:
+			columns[shape.Name] = make([]int32, length)
+		case io.INT64:
+			columns[shape.Name] = make([]int64, length)
+		case io.FLOAT32:
+			columns[shape.Name] = make([]float32, length)
+		case io.FLOAT64:
+			columns[shape.Name] = make([]float64, length)
+		default:
+			panic(fmt.Sprintf("unsupported shape: %v", shape.Type))
+		}
+	}
+	for i, entry := range p.Entries {
+		row := map[string]interface{}{}
+		if err := json.Unmarshal(entry.Data, &row); err != nil {
+			return nil, fmt.Errorf("Failed to unmarshal slait publication entry to bar - Error: %v", err)
+		}
+		for name, data := range row {
+			// var value reflect.Value
+			var v interface{}
+			if str, ok := data.(string); ok {
+				name = "Epoch"
+				t, err := time.Parse(time.RFC3339, str)
+				if err != nil {
+					return nil, err
+				}
+				v = reflect.ValueOf(t.Unix()).Interface()
+			} else {
+				v = reflect.ValueOf(data).Interface()
+			}
+			value := reflect.ValueOf(columns[name])
+			e := value.Index(i)
+			e.Set(reflect.ValueOf(v))
+		}
+	}
+	cs := io.NewColumnSeries()
+	for name, data := range columns {
+		cs.AddColumn(name, data)
+	}
+	csm := io.NewColumnSeriesMap()
+	tbk := io.NewTimeBucketKey(fmt.Sprintf("%v/1Min/%v", p.Partition, ss.attributeGroup))
+	csm.AddColumnSeries(*tbk, cs)
+	return &csm, nil
+}
+
+func (ss *SlaitSubscriber) read() (err error) {
+	defer func() {
+		ss.done <- struct{}{}
+		close(ss.done)
+	}()
+	for {
+		msgType, msg, err := ss.conn.ReadMessage()
+		if err != nil {
+			glog.Errorf("Failed to read message from Slait - Error: %v", err)
+			return err
+		}
+		err = ss.handleMessage(msg, msgType)
+		if err != nil {
+			return err
```

This error is not handled at all

---

ya missed a log there.